replit_river/transport.py

```diff
@@ -45,13 +45,22 @@ async def _close_all_sessions(self) -> None:
         await asyncio.gather(*tasks)
         logging.info(f"Transport closed {self._transport_id}")
 
-    async def _delete_session(self, session: Session) -> None:
-        async with self._session_lock:
+    async def _delete_session(
+        self, session: Session, acquire_lock: bool = True
+    ) -> None:
+        if acquire_lock:
+            async with self._session_lock:
```

Comment:
why do we need a lock for these two?

Reply:
Actually not needed in the context of asyncio